### DISCUSSION TITLE
Match token but don't consume it?

### DISCUSSION THREAD

**Discussion by rowlesmr** [lark-parser/lark#1162] [Q&A]
How can I describe a rule that must match a series of tokens to succeed, but leave the last rule to be consumed by a separate rule?

eg:
```
comment: ('#' anyprintchar* eol)+
whitespace: (" " | "\t" | eol | comment)+
singlequotedstring: "'" anyprintchar* "'" whitespace
```

Here, `singlequotedstring` must have whitespace after the final quote mark, but that whitespace is not part of the quoted string.

**Comment by erezsh** [comment]
There is no syntax in Lark for this behavior. But you can emulate it yourself if you use `Lark.parse_interactive()`.
